build: relocate toolchain-tools to usr/local

edymtt opened the pull request:
Supports rdar://problem/53670881

edymtt committed:
build: relocate toolchain-tools to usr/local

Supports rdar://problem/53670881

edymtt commented:
@swift-ci test

edymtt commented:
@swift-ci Please Build Toolchain macOS Platform

edymtt commented:
@swift-ci Please Build Toolchain Linux Platform

edymtt commented:
@swift-ci please test windows

edymtt commented:
Followup to #28239

edymtt commented on cmake/modules/AddSwift.cmake:
```diff
@@ -2423,7 +2423,7 @@ function(add_swift_host_tool executable)
   swift_install_in_component(TARGETS ${executable}
                              RUNTIME
-                               DESTINATION bin
+                               DESTINATION "${ASHT_DESTINATION_PREFIX}bin"
```
I'm not entirely sure if this is an idiomatic enough way to compose path in CMake -- I've tried exploring `get_filename_component` and `file(TO_NATIVE_PATH )`, but none of those seems to fit this scenario (prepend a folder to a relative path)

nkcsgexi replied:
Can we have a boolean parameter called `IS_LOCAL` rather than asking the called to pass down the prefix? this may lead to the duplications of the prefix.

edymtt replied:
Sure -- I've made this change in d70625f820a7475093c2fcd0baae72016077c9b4

compnerd replied:
I really am against the idea of adding more parameters to `swift_install_in_component`.  This function is needlessly complicated and deviates from the standard practices for CMake, making this codebase more difficult for new comers.  I think that it is better to use this as CMake does, and we can absolutely just use `local/bin` as the destination.

edymtt committed:
Simplify relocation of swift-api-digester

As per review feedback, instead of requiring the user to pass down the
prefix, provide a boolean parameter to opt in for relocation into
usr/local

edymtt commented:
@swift-ci please test Windows

edymtt commented:
@swift-ci test

edymtt commented:
@swift-ci Please Build Toolchain macOS Platform

edymtt commented:
@swift-ci Please Build Toolchain Linux Platform

edymtt commented:
@swift-ci please test Windows

edymtt commented:
@swift-ci please test Windows

compnerd commented:
The local/bin is definitely an issue for Windows - I don't think that we should be emulating that path.  Why do we even want to install the test binaries?  Adding more complexity to the custom wrappers is the opposite of where the build should be going IMO - we want to simplify (and in fact, this was a discussion at the LLVM developer conference in regards to LLVM too).

futurejones commented:
Can one of the admins verify this patch?

shahmishal commented:
Please update the base branch to `main` by Oct 5th otherwise the pull request will be closed automatically. 

* How to change the base branch: ([Link](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/changing-the-base-branch-of-a-pull-request))
* More detail about the branch update: ([Link](https://forums.swift.org/t/updating-branch-names/40412/))
